WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift

```diff
@@ -11,13 +12,22 @@ class ProductListViewModel {
 
     let siteID: Int64
     private let stores: StoresManager
+    private let storage: StorageManagerType
 
     private(set) var selectedProducts: Set<Product> = .init()
 
+    private var wooSubscriptionProductsEligibilityChecker: WooSubscriptionProductsEligibilityCheckerProtocol
+
     init(siteID: Int64,
-         stores: StoresManager = ServiceLocator.stores) {
+         stores: StoresManager = ServiceLocator.stores,
+         storage: StorageManagerType = ServiceLocator.storageManager) {
         self.siteID = siteID
         self.stores = stores
+        self.storage = storage
+        self.wooSubscriptionProductsEligibilityChecker = WooSubscriptionProductsEligibilityChecker(
+            siteID: siteID,
+            storage: storage
+        )
```

Comment:
I see that you aren't using the `storage` to write unit tests for `ProductListViewModel`. 

- In this case you can skip injecting it into `ProductListViewModel` class.
- You can just pass the `siteID` to `WooSubscriptionProductsEligibilityChecker`'s init. 

```suggestion
        self.wooSubscriptionProductsEligibilityChecker = WooSubscriptionProductsEligibilityChecker(siteID: siteID)
```

Reply:
Handled in https://github.com/woocommerce/woocommerce-ios/pull/11236/commits/108f60ac9d727cc958f44efadd6810eb36800c91